protocol/linkpreview/linkpreview.go

```diff
@@ -149,9 +162,38 @@ func (u OEmbedUnfurler) newOEmbedURL() (*neturl.URL, error) {
 	return oembedURL, nil
 }
 
+func handlePhotoOembedType(preview *common.LinkPreview, response OEmbedBaseResponse) {
+	if response.URL != "" {
+		preview.Thumbnail.URL = response.URL
+	}
+}
+
+func handleVideoOembedType(preview *common.LinkPreview, response OEmbedBaseResponse) {
+	if response.ThumbnailURL != "" {
+		preview.Thumbnail.URL = response.ThumbnailURL
+	}
+	if response.Width != 0 {
+		preview.Thumbnail.Width = response.ThumbnailWidth
+	}
+	if response.Height != 0 {
+		preview.Thumbnail.Height = response.ThumbnailHeight
+	}
+}
+
+func handleRichOembedType(preview *common.LinkPreview, response OEmbedBaseResponse) {
+	if response.ThumbnailURL != "" {
```

Comment:
It seems all these checks could be removed, also in the other `handle*` functions. I changed the code locally and the tests pass (except for the failure which you're aware of in the CI as well).

The zero-values of the response object are basically empty strings or zero for numbers so we can just always set struct field anyway. 

To give you an example for this `response.ThumbnailURL != ""` check (but the reasoning should hold for the other checks):

`preview.Thumbnail.URL` is initialized as the zero-value of the string, thus empty. Checking if `response.ThumbnailURL != ""` is unnecessary because if we set `preview.Thumbnail.URL = response.ThumbnailURL` it will be the empty string, just as the zero value of the struct field. This same logic could be used for the other checks.

Maybe I'm missing something.

Reply:
Removed, So I get it nil is the ones that sometimes is problematic, That's why we tend to check for it.

Reply:
I thought a `string` struct field cannot be nil, so there's never a case where `OEmbedBaseResponse` has nil values.